src/platform.rs

```diff
@@ -66,6 +74,15 @@ pub fn new_default_platform(
   Platform::new(thread_pool_size, idle_task_support)
 }
 
+/// TODO: doc comment
+#[inline(always)]
```

Comment:
```suggestion
/// Creates a platform that is identical to the default platform, but does not enforce
/// thread-isolated allocations. This may reduce security in some cases, so this method
/// should be used with caution in cases where the threading guarantees of
/// `new_default_platform` cannot be upheld (generally for tests).
#[inline(always)]
```